### PR TITLE
doc/dev/index.rst: begin writing Contributing to Ceph

### PR DESCRIPTION
**Pull request by smithfarm**
Replace existing index.rst with the beginnings of a comprehensive developer guide.

Move the old index.rst to index-old.rst.

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: doc/dev/index.rst [hunk 1]
```diff
@@ -27,13 +80,102 @@ a message to ``majordomo@vger.kernel.org`` with the line::
...
+
+Ceph supports a number of major Linux distributions and provides packaging for them.
+
+Building
+========
+
+Building from source
+--------------------
+
+ceph ceph ceph
```
Comment (dachary) [inline doc/dev/index.rst]: A link to http://docs.ceph.com/docs/master/install/build-ceph/

**Comment by dachary** [comment]
With the above link it's a nice first step :-)

<pre>
Reviewed-by: Loic Dachary &lt;ldachary@redhat.com>
</pre>

**Comment by smithfarm** [comment]
Link added.

**Comment by smithfarm** [comment]
@dachary I got a Dockerized test environment running. Testing uncovered two bugs in the new `index.rst` which are now fixed.

**Pull request merged**